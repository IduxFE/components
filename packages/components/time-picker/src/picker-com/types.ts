export interface TimePickerPanelProps {
  value?: Date
  defaultOpenValue?: Date
  hourStep: number
  minuteStep: number
  secondStep: number
  disabledHours: (selectAmPm: string) => number[]
  disabledMinutes: (selectedHour: number, selectAmPm: string) => number[]
  disabledSeconds: (selectedHour: number, selectedMinute: number, selectAmPm: string) => number[]
  hideDisabledOptions: boolean
  showSelects: boolean[]
  amPmCapital: boolean
}

export interface TimeOptionsProps {
  value: number | string
  options: TimeOption[]
}
export interface TimeOption {
  value: number | string
  disabled: boolean
}

export type SelectType = 'hour' | 'minute' | 'second' | 'ampm'
