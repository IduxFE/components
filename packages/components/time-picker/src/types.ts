import { AbstractControl } from '@idux/cdk/forms'
import { TimePickerSize } from '@idux/components/core/config'
import type { DefineComponent } from 'vue'

interface CommonProps {
  control?: string | AbstractControl
  format: string
  clearable?: boolean
  autofocus?: boolean
  borderless?: boolean
  suffix?: string
  clearIcon?: string
  clearText?: string
  size?: TimePickerSize
  open: boolean
  disabledHours?: (selectedAmPm: string) => number[]
  disabledMinutes?: (selectedHour: number, selectedAmPm: string) => number[]
  disabledSeconds?: (selectedHour: number, selectedMinute: number, selectedAmPm: string) => number[]
  hideDisabledOptions?: boolean
  hourStep?: number
  minuteStep?: number
  secondStep?: number
  overlayClassName?: string
}

export interface TimePickerProps extends CommonProps {
  value?: Date
  placeholder: string
  disabled: boolean
  readonly: boolean
  defaultOpenValue: Date
}

export interface TimeRangePickerProps extends CommonProps {
  value?: [Date, Date]
  placeholder: [string, string]
  disabled: boolean | [boolean, boolean]
  readonly: boolean | [boolean, boolean]
  defaultOpenValue: [Date, Date]
  separator: string
}

export type TimePickerComponent = InstanceType<DefineComponent<TimePickerProps>>
