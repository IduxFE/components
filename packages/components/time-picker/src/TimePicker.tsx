import { defineComponent, computed } from 'vue'
import { TimePickerProps } from './types'
import { IxPropTypes } from '@idux/cdk/utils'
import { controlProp } from '@idux/cdk/forms'

import TimePickerPanel from './picker-com/TimePickerPanel.vue'
import { IxIcon } from '@idux/components/icon'
import { IxInput } from '@idux/components/input'
import { IxPortal } from '@idux/cdk/portal'
import { clickOutside } from '@idux/cdk/click-outside'
import { IxTypography } from '@idux/components/typography'

import { object } from 'vue-types'
import { useCommonBindings } from './useCommonBinding'
import { usePickerOverlay } from './usePickerOverlay'
import { useCommonProps } from './useCommonProps'
import { useClearIcon } from './useClearIcon'
import dayjs from 'dayjs'
import { normalizeFormat } from './utils'
import { useGlobalConfig } from '@idux/components/config'

export default defineComponent({
  name: 'IxTimePicker',
  components: {
    IxInput,
    IxIcon,
    IxPortal,
    TimePickerPanel,
  },
  directives: {
    clickOutside,
    typography: IxTypography,
  },
  props: {
    control: controlProp,
    value: object<Date>(),
    defaultOpenValue: object<Date>().def(() => dayjs().hour(0).minute(0).second(0).toDate()),

    format: IxPropTypes.string.def('HH:mm:ss'),
    placeholder: IxPropTypes.string.def('请选择时间'),
    disabled: IxPropTypes.bool.def(false),
    readonly: IxPropTypes.bool.def(false),
    clearable: IxPropTypes.bool,
    autofocus: IxPropTypes.bool,
    borderless: IxPropTypes.bool.def(false),
    suffix: IxPropTypes.string,
    clearIcon: IxPropTypes.string,
    clearText: IxPropTypes.string,
    size: IxPropTypes.oneOf(['large', 'medium', 'small'] as const),

    open: IxPropTypes.bool.def(false),
    overlayClassName: IxPropTypes.string,
    hourStep: IxPropTypes.number,
    minuteStep: IxPropTypes.number,
    secondStep: IxPropTypes.number,
    disabledHours: IxPropTypes.func,
    disabledMinutes: IxPropTypes.func,
    disabledSeconds: IxPropTypes.func,
    hideDisabledOptions: IxPropTypes.bool,
  },
  emits: ['update:value', 'update:open', 'change', 'clear', 'focus', 'blur'],
  setup(props: TimePickerProps, { slots }) {
    const globalConfig = useGlobalConfig('timePicker')

    const { onFocus, onBlur, onClear, onValueChange, valueAccessor } = useCommonBindings()

    const { overlayTriggerRef, overlayRef, overlayVisibility, hideOverlay, showOverlay } = usePickerOverlay(props)

    const { pickerProps, inputProps } = useCommonProps(props, globalConfig)

    const {
      onMouseEnter,
      onMouseLeave,
      showClearIcon,
      clearText: clearIconText,
      clearIcon: clearIconName,
    } = useClearIcon(props, globalConfig)

    const inputValue = computed(() =>
      valueAccessor.value ? dayjs(valueAccessor.value as Date).format(normalizeFormat(props.format)) : '',
    )

    function clearValue(evt: MouseEvent) {
      onClear(evt)
      hideOverlay()
    }

    function renderClearIcon() {
      return (<span class="ix-time-picker__clear-icon" onClick={clearValue}>
        {slots.clearIcon?.() ?? <ix-icon name={clearIconName} title={clearIconText} />}
      </span>)
    }

    function renderSuffix() {
      return (slots.suffix?.() ?? (<ix-icon name={inputProps.suffix} />)
      )
    }

return () => ((
  <div
    ref="overlayTriggerRef"
    v-click-outside={hideOverlay}
    class="ix-time-picker"
    onMouseenter={onMouseEnter}
    onMouseleave={onMouseLeave}
  >
    <ix-input
      v-bind={inputProps}
      value={inputValue}
      onFocus={onFocus}
      onBlur={onBlur}
      onKeyup_esc={hideOverlay}
      onKeyup_enter={showOverlay}
    >
      <span slot="suffix">
        {showClearIcon ? renderClearIcon() : renderSuffix()}
      </span>
    </ix-input>

    <ix-portal target="ix-time-picker-container">
      <transition name="ix-fade">
        <time-picker-panel
          v-show={overlayVisibility}
          ref="overlayRef"
          class={props.overlayClassName}
          v-bind={pickerProps}
          onChange={onValueChange}
        />
      </transition>
    </ix-portal>
  </div>
)

    // return {
    //   onFocus,
    //   onBlur,
    //   onValueChange,
    //   inputValue,
    //   inputProps,
    //   pickerProps,
    //   clearValue,

    //   // about clearIcon
    //   onMouseEnter,
    //   onMouseLeave,
    //   showClearIcon,
    //   clearIconText,
    //   clearIconName,

    //   // about overlay
    //   overlayTriggerRef,
    //   overlayRef,
    //   overlayVisibility,
    //   hideOverlay,
    //   showOverlay,
    // }
  }
})
