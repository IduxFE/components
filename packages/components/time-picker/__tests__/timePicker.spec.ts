import { mount, VueWrapper } from '@vue/test-utils'
import { renderWork, wait } from '@tests'
import IxTimePicker from '../src/TimePicker.vue'

import { TimePickerProps } from '../src/types'
import { ComponentPublicInstance, nextTick, reactive, UnwrapRef } from 'vue'
import dayjs from 'dayjs'

describe('TimePicker.vue', () => {
  let wrapper: VueWrapper<ComponentPublicInstance>
  let pickerContainer: HTMLElement
  let pickerPanel: HTMLElement

  function getOptionList(): NodeListOf<HTMLElement> {
    return pickerPanel.querySelectorAll('.ix-time-options')
  }

  function getOptionsItemList(listIndex: number): HTMLElement[] {
    return Array.from(getOptionList()[listIndex].children) as HTMLElement[]
  }

  function getSelectedIndex(listIndex: number): number {
    return getOptionsItemList(listIndex).findIndex(item => item.classList.contains('ix-time-options__item--selected'))
  }

  function getDisabledIndexs(listIndex: number): number[] {
    return getOptionsItemList(listIndex)
      .map((item, index) => ({
        index,
        cls: item.classList,
      }))
      .filter(item => item.cls.contains('ix-time-options__item--disabled'))
      .map(item => item.index)
  }

  function clickOption(listIndex: number, optionIndex: number) {
    getOptionsItemList(listIndex)[optionIndex].click()
  }

  function clickClearIcon() {
    ;(document.querySelector('.ix-time-picker__clear-icon') as HTMLElement)?.click()
  }

  async function initWrapper(
    props: UnwrapRef<Partial<TimePickerProps>>,
    listeners: Record<string, (...args: unknown[]) => unknown> = {},
  ) {
    wrapper = mount(
      {
        components: { IxTimePicker },
        template: `
      <ix-time-picker 
        v-model:value="props.value" 
        v-model:open="props.open"
        v-bind="props"
        v-on="listeners"
      />`,
        setup() {
          return {
            props,
            listeners,
          }
        },
      },
      { attachTo: 'body' },
    )

    await nextTick()

    pickerContainer = document.querySelector('.ix-time-picker-container')!
    pickerPanel = pickerContainer.querySelector('.ix-time-picker-panel')!
  }

  beforeEach(() => {
    if (pickerContainer) {
      pickerContainer.innerHTML = ''
    }
  })

  renderWork(IxTimePicker)

  test('v-model:value work', async () => {
    const props = reactive({
      value: dayjs().hour(1).minute(2).second(3).toDate(),
      open: true,
      // 24hours
      format: 'HH:mm:ss',
    })
    await initWrapper(props)

    // test initial value
    expect(getSelectedIndex(0)).toBe(1)
    expect(getSelectedIndex(1)).toBe(2)
    expect(getSelectedIndex(2)).toBe(3)

    // test value change
    props.value = dayjs().hour(10).minute(20).second(30).toDate()
    await nextTick()
    expect(getSelectedIndex(0)).toBe(10)
    expect(getSelectedIndex(1)).toBe(20)
    expect(getSelectedIndex(2)).toBe(30)

    // test click change value
    clickOption(0, 5)
    clickOption(1, 5)
    clickOption(2, 5)
    expect(props.value.getHours()).toBe(5)
    expect(props.value.getMinutes()).toBe(5)
    expect(props.value.getSeconds()).toBe(5)

    /** test ampm */
    props.format = 'hh:mm:ss a'

    // 12 am
    props.value = dayjs().hour(0).toDate()
    await nextTick()
    expect(getSelectedIndex(0)).toBe(0)
    expect(getSelectedIndex(3)).toBe(0)

    // 10 am
    props.value = dayjs().hour(10).toDate()
    await nextTick()
    expect(getSelectedIndex(0)).toBe(10)
    expect(getSelectedIndex(3)).toBe(0)

    // 12 pm
    props.value = dayjs().hour(12).toDate()
    await nextTick()
    expect(getSelectedIndex(0)).toBe(0)
    expect(getSelectedIndex(3)).toBe(1)

    // 3 pm
    props.value = dayjs().hour(15).toDate()
    await nextTick()
    expect(getSelectedIndex(0)).toBe(3)
    expect(getSelectedIndex(3)).toBe(1)

    // 12 am
    props.value = dayjs().hour(24).toDate()
    await nextTick()
    expect(getSelectedIndex(0)).toBe(0)
    expect(getSelectedIndex(3)).toBe(0)

    // test click change value when use ampm
    props.value = dayjs().hour(5).minute(5).second(5).toDate()
    await nextTick()

    clickOption(0, 0)
    expect(props.value.getHours()).toBe(0)
    expect(props.value.getMinutes()).toBe(5)
    expect(props.value.getSeconds()).toBe(5)

    clickOption(3, 1)
    expect(props.value.getHours()).toBe(12)
    expect(props.value.getMinutes()).toBe(5)
    expect(props.value.getSeconds()).toBe(5)

    clickOption(0, 5)
    expect(props.value.getHours()).toBe(17)
    expect(props.value.getMinutes()).toBe(5)
    expect(props.value.getSeconds()).toBe(5)

    clickOption(3, 0)
    expect(props.value.getHours()).toBe(5)
    expect(props.value.getMinutes()).toBe(5)
    expect(props.value.getSeconds()).toBe(5)

    // test clear value
    await wrapper.findComponent(IxTimePicker).trigger('mouseenter')
    clickClearIcon()
    expect(props.value).toBe(null)
  })

  test('v-model:open work', async () => {
    const props = reactive({ open: true })
    await initWrapper(props)

    expect(pickerPanel.style.display).toBe('')

    props.open = false
    await wait(600)
    expect(pickerPanel.style.display).toBe('none')

    await wrapper.find('.ix-time-picker').trigger('click')
    await wait(200)
    expect(pickerPanel.style.display).toBe('')
    expect(props.open).toBe(true)
  })

  test('format work', async () => {
    const props = reactive<Partial<TimePickerProps>>({
      value: dayjs().hour(1).minute(1).second(1).toDate(),
    })
    await initWrapper(props)

    props.format = 'hh:mm:ss'
    await nextTick()
    expect(pickerPanel.innerHTML).toMatchSnapshot()

    props.format = 'hh:mm:ss a'
    await nextTick()
    expect(pickerPanel.innerHTML).toMatchSnapshot()
  })

  test('disabled & disabledOptions work', async () => {
    const props = reactive<Partial<TimePickerProps>>({
      value: dayjs().hour(0).minute(0).second(0).toDate(),
      open: true,
    })
    await initWrapper(props)

    // test input disabled
    props.disabled = true
    await nextTick()
    expect(wrapper.find('.ix-input').classes()).toContain('ix-input-disabled')

    props.disabled = false
    await nextTick()
    expect(wrapper.find('.ix-input').classes()).not.toContain('ix-input-disabled')

    // test disabled fn
    props.disabledHours = () => [1, 2, 3]
    props.disabledMinutes = (hour: number) => (hour === 5 ? [1, 2, 3] : [])
    props.disabledSeconds = (hour: number, minute: number) => (hour === 5 && minute === 5 ? [1, 2, 3] : [])
    props.value = dayjs().hour(5).minute(5).second(0).toDate()
    await nextTick()
    expect(getDisabledIndexs(0)).toEqual([1, 2, 3])
    expect(getDisabledIndexs(1)).toEqual([1, 2, 3])
    expect(getDisabledIndexs(2)).toEqual([1, 2, 3])

    // test clicking disabled options
    clickOption(0, 1)
    clickOption(1, 1)
    clickOption(2, 1)
    expect(props.value.getHours()).toBe(5)
    expect(props.value.getMinutes()).toBe(5)
    expect(props.value.getSeconds()).toBe(0)

    // test hiding disabled options
    props.hideDisabledOptions = true
    await nextTick()
    expect(getOptionsItemList(0).length).toBe(21)
    expect(getOptionsItemList(1).length).toBe(57)
    expect(getOptionsItemList(2).length).toBe(57)

    expect(getDisabledIndexs(0).length).toBe(0)
    expect(getDisabledIndexs(1).length).toBe(0)
    expect(getDisabledIndexs(2).length).toBe(0)
  })

  test('clearable & clearIcon & clearText work', async () => {
    const props = reactive<Partial<TimePickerProps>>({
      value: new Date(),
    })
    await initWrapper(props)

    const commonIconCls = '.ix-time-picker__clear-icon'
    const defaultIconCls = `${commonIconCls} .ix-icon-close-circle`

    // test hover
    await wrapper.findComponent(IxTimePicker).trigger('mouseenter')
    expect(wrapper.find(defaultIconCls).exists()).toBe(true)

    await wrapper.findComponent(IxTimePicker).trigger('mouseleave')
    expect(wrapper.find(defaultIconCls).exists()).toBe(false)

    // test clearable false
    props.clearable = false
    await wrapper.findComponent(IxTimePicker).trigger('mouseenter')
    expect(wrapper.find(defaultIconCls).exists()).toBe(false)

    // test clearIcon and clearText
    props.clearable = true
    props.clearIcon = 'close'
    props.clearText = 'clearTextToChange'
    await wrapper.findComponent(IxTimePicker).trigger('mouseenter')
    expect(wrapper.find(`${commonIconCls} .ix-icon-close`).exists()).toBe(true)
    expect(wrapper.find(`${commonIconCls} .ix-icon-close`).attributes('title')).toBe('clearTextToChange')
  })

  test('focus and blur work', async () => {
    const props = reactive({
      value: new Date(),
    })
    const focusCb = jest.fn()
    const blurCb = jest.fn()

    await initWrapper(props, {
      focus: focusCb,
      blur: blurCb,
    })

    await wrapper.find('input').trigger('focus')
    expect(focusCb).toBeCalled()

    await wrapper.find('input').trigger('blur')
    expect(blurCb).toBeCalled()
  })

  test('suffix work', async () => {
    const props = reactive<Partial<TimePickerProps>>({})
    await initWrapper(props)

    props.suffix = 'calendar'
    await nextTick()
    expect(wrapper.find(`.ix-input-suffix .ix-icon-calendar`).exists()).toBe(true)
  })

  test('optionsStep work', async () => {
    const props = reactive<Partial<TimePickerProps>>({})
    await initWrapper(props)

    function getOptionsContent(listIndex: number) {
      return getOptionsItemList(listIndex).map(item => Number(item.innerHTML))
    }

    // if every option equal the last option plus step
    function isStepWork(list: number[], step = 1): boolean {
      let worked = true

      list.forEach((item, idx) => {
        if (idx > 0 && item !== list[idx - 1] + step) {
          worked = false
        }
      })

      return worked
    }

    props.hourStep = 5
    props.minuteStep = 7
    props.secondStep = 7
    await nextTick()
    expect(isStepWork(getOptionsContent(0), 5)).toEqual(true)
    expect(isStepWork(getOptionsContent(1), 7)).toEqual(true)
    expect(isStepWork(getOptionsContent(2), 7)).toEqual(true)
  })

  test('defaultOpenValue work', async () => {
    const props = reactive<Partial<TimePickerProps>>({})
    await initWrapper(props)

    await wrapper.findComponent(IxTimePicker).trigger('click')
    expect(getSelectedIndex(0)).toBe(0)
    expect(getSelectedIndex(1)).toBe(0)
    expect(getSelectedIndex(2)).toBe(0)

    props.defaultOpenValue = dayjs().hour(12).minute(20).second(30).toDate()
    await nextTick()
    expect(getSelectedIndex(0)).toBe(12)
    expect(getSelectedIndex(1)).toBe(20)
    expect(getSelectedIndex(2)).toBe(30)

    props.format = 'hh:mm:ss a'
    await nextTick()
    expect(getSelectedIndex(0)).toBe(0)
    expect(getSelectedIndex(1)).toBe(20)
    expect(getSelectedIndex(2)).toBe(30)

    props.defaultOpenValue = dayjs().hour(15).minute(20).second(30).toDate()
    await nextTick()
    expect(getSelectedIndex(0)).toBe(3)
    expect(getSelectedIndex(1)).toBe(20)
    expect(getSelectedIndex(2)).toBe(30)

    props.value = dayjs().hour(5).minute(5).second(5).toDate()
    await nextTick()
    expect(getSelectedIndex(0)).toBe(5)
    expect(getSelectedIndex(1)).toBe(5)
    expect(getSelectedIndex(2)).toBe(5)
  })

  test('overlayClassName work', async () => {
    const props = reactive<Partial<TimePickerProps>>({
      overlayClassName: 'test-overlay-class-name',
    })
    await initWrapper(props)

    expect(pickerPanel.classList).toContain('test-overlay-class-name')
  })
})
