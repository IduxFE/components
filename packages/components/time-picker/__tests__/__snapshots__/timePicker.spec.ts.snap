// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePicker.vue format work 1`] = `"<ul class=\\"ix-time-options\\"><li class=\\"ix-time-options__item\\">12</li><li class=\\"ix-time-options__item ix-time-options__item--selected\\">01</li><li class=\\"ix-time-options__item\\">02</li><li class=\\"ix-time-options__item\\">03</li><li class=\\"ix-time-options__item\\">04</li><li class=\\"ix-time-options__item\\">05</li><li class=\\"ix-time-options__item\\">06</li><li class=\\"ix-time-options__item\\">07</li><li class=\\"ix-time-options__item\\">08</li><li class=\\"ix-time-options__item\\">09</li><li class=\\"ix-time-options__item\\">10</li><li class=\\"ix-time-options__item\\">11</li></ul><ul class=\\"ix-time-options\\"><li class=\\"ix-time-options__item\\">00</li><li class=\\"ix-time-options__item ix-time-options__item--selected\\">01</li><li class=\\"ix-time-options__item\\">02</li><li class=\\"ix-time-options__item\\">03</li><li class=\\"ix-time-options__item\\">04</li><li class=\\"ix-time-options__item\\">05</li><li class=\\"ix-time-options__item\\">06</li><li class=\\"ix-time-options__item\\">07</li><li class=\\"ix-time-options__item\\">08</li><li class=\\"ix-time-options__item\\">09</li><li class=\\"ix-time-options__item\\">10</li><li class=\\"ix-time-options__item\\">11</li><li class=\\"ix-time-options__item\\">12</li><li class=\\"ix-time-options__item\\">13</li><li class=\\"ix-time-options__item\\">14</li><li class=\\"ix-time-options__item\\">15</li><li class=\\"ix-time-options__item\\">16</li><li class=\\"ix-time-options__item\\">17</li><li class=\\"ix-time-options__item\\">18</li><li class=\\"ix-time-options__item\\">19</li><li class=\\"ix-time-options__item\\">20</li><li class=\\"ix-time-options__item\\">21</li><li class=\\"ix-time-options__item\\">22</li><li class=\\"ix-time-options__item\\">23</li><li class=\\"ix-time-options__item\\">24</li><li class=\\"ix-time-options__item\\">25</li><li class=\\"ix-time-options__item\\">26</li><li class=\\"ix-time-options__item\\">27</li><li class=\\"ix-time-options__item\\">28</li><li class=\\"ix-time-options__item\\">29</li><li class=\\"ix-time-options__item\\">30</li><li class=\\"ix-time-options__item\\">31</li><li class=\\"ix-time-options__item\\">32</li><li class=\\"ix-time-options__item\\">33</li><li class=\\"ix-time-options__item\\">34</li><li class=\\"ix-time-options__item\\">35</li><li class=\\"ix-time-options__item\\">36</li><li class=\\"ix-time-options__item\\">37</li><li class=\\"ix-time-options__item\\">38</li><li class=\\"ix-time-options__item\\">39</li><li class=\\"ix-time-options__item\\">40</li><li class=\\"ix-time-options__item\\">41</li><li class=\\"ix-time-options__item\\">42</li><li class=\\"ix-time-options__item\\">43</li><li class=\\"ix-time-options__item\\">44</li><li class=\\"ix-time-options__item\\">45</li><li class=\\"ix-time-options__item\\">46</li><li class=\\"ix-time-options__item\\">47</li><li class=\\"ix-time-options__item\\">48</li><li class=\\"ix-time-options__item\\">49</li><li class=\\"ix-time-options__item\\">50</li><li class=\\"ix-time-options__item\\">51</li><li class=\\"ix-time-options__item\\">52</li><li class=\\"ix-time-options__item\\">53</li><li class=\\"ix-time-options__item\\">54</li><li class=\\"ix-time-options__item\\">55</li><li class=\\"ix-time-options__item\\">56</li><li class=\\"ix-time-options__item\\">57</li><li class=\\"ix-time-options__item\\">58</li><li class=\\"ix-time-options__item\\">59</li></ul><ul class=\\"ix-time-options\\"><li class=\\"ix-time-options__item\\">00</li><li class=\\"ix-time-options__item ix-time-options__item--selected\\">01</li><li class=\\"ix-time-options__item\\">02</li><li class=\\"ix-time-options__item\\">03</li><li class=\\"ix-time-options__item\\">04</li><li class=\\"ix-time-options__item\\">05</li><li class=\\"ix-time-options__item\\">06</li><li class=\\"ix-time-options__item\\">07</li><li class=\\"ix-time-options__item\\">08</li><li class=\\"ix-time-options__item\\">09</li><li class=\\"ix-time-options__item\\">10</li><li class=\\"ix-time-options__item\\">11</li><li class=\\"ix-time-options__item\\">12</li><li class=\\"ix-time-options__item\\">13</li><li class=\\"ix-time-options__item\\">14</li><li class=\\"ix-time-options__item\\">15</li><li class=\\"ix-time-options__item\\">16</li><li class=\\"ix-time-options__item\\">17</li><li class=\\"ix-time-options__item\\">18</li><li class=\\"ix-time-options__item\\">19</li><li class=\\"ix-time-options__item\\">20</li><li class=\\"ix-time-options__item\\">21</li><li class=\\"ix-time-options__item\\">22</li><li class=\\"ix-time-options__item\\">23</li><li class=\\"ix-time-options__item\\">24</li><li class=\\"ix-time-options__item\\">25</li><li class=\\"ix-time-options__item\\">26</li><li class=\\"ix-time-options__item\\">27</li><li class=\\"ix-time-options__item\\">28</li><li class=\\"ix-time-options__item\\">29</li><li class=\\"ix-time-options__item\\">30</li><li class=\\"ix-time-options__item\\">31</li><li class=\\"ix-time-options__item\\">32</li><li class=\\"ix-time-options__item\\">33</li><li class=\\"ix-time-options__item\\">34</li><li class=\\"ix-time-options__item\\">35</li><li class=\\"ix-time-options__item\\">36</li><li class=\\"ix-time-options__item\\">37</li><li class=\\"ix-time-options__item\\">38</li><li class=\\"ix-time-options__item\\">39</li><li class=\\"ix-time-options__item\\">40</li><li class=\\"ix-time-options__item\\">41</li><li class=\\"ix-time-options__item\\">42</li><li class=\\"ix-time-options__item\\">43</li><li class=\\"ix-time-options__item\\">44</li><li class=\\"ix-time-options__item\\">45</li><li class=\\"ix-time-options__item\\">46</li><li class=\\"ix-time-options__item\\">47</li><li class=\\"ix-time-options__item\\">48</li><li class=\\"ix-time-options__item\\">49</li><li class=\\"ix-time-options__item\\">50</li><li class=\\"ix-time-options__item\\">51</li><li class=\\"ix-time-options__item\\">52</li><li class=\\"ix-time-options__item\\">53</li><li class=\\"ix-time-options__item\\">54</li><li class=\\"ix-time-options__item\\">55</li><li class=\\"ix-time-options__item\\">56</li><li class=\\"ix-time-options__item\\">57</li><li class=\\"ix-time-options__item\\">58</li><li class=\\"ix-time-options__item\\">59</li></ul><ul class=\\"ix-time-options\\"><li class=\\"ix-time-options__item ix-time-options__item--selected\\">am</li><li class=\\"ix-time-options__item\\">pm</li></ul>"`;

exports[`TimePicker.vue format work 2`] = `"<ul class=\\"ix-time-options\\"><li class=\\"ix-time-options__item\\">12</li><li class=\\"ix-time-options__item ix-time-options__item--selected\\">01</li><li class=\\"ix-time-options__item\\">02</li><li class=\\"ix-time-options__item\\">03</li><li class=\\"ix-time-options__item\\">04</li><li class=\\"ix-time-options__item\\">05</li><li class=\\"ix-time-options__item\\">06</li><li class=\\"ix-time-options__item\\">07</li><li class=\\"ix-time-options__item\\">08</li><li class=\\"ix-time-options__item\\">09</li><li class=\\"ix-time-options__item\\">10</li><li class=\\"ix-time-options__item\\">11</li></ul><ul class=\\"ix-time-options\\"><li class=\\"ix-time-options__item\\">00</li><li class=\\"ix-time-options__item ix-time-options__item--selected\\">01</li><li class=\\"ix-time-options__item\\">02</li><li class=\\"ix-time-options__item\\">03</li><li class=\\"ix-time-options__item\\">04</li><li class=\\"ix-time-options__item\\">05</li><li class=\\"ix-time-options__item\\">06</li><li class=\\"ix-time-options__item\\">07</li><li class=\\"ix-time-options__item\\">08</li><li class=\\"ix-time-options__item\\">09</li><li class=\\"ix-time-options__item\\">10</li><li class=\\"ix-time-options__item\\">11</li><li class=\\"ix-time-options__item\\">12</li><li class=\\"ix-time-options__item\\">13</li><li class=\\"ix-time-options__item\\">14</li><li class=\\"ix-time-options__item\\">15</li><li class=\\"ix-time-options__item\\">16</li><li class=\\"ix-time-options__item\\">17</li><li class=\\"ix-time-options__item\\">18</li><li class=\\"ix-time-options__item\\">19</li><li class=\\"ix-time-options__item\\">20</li><li class=\\"ix-time-options__item\\">21</li><li class=\\"ix-time-options__item\\">22</li><li class=\\"ix-time-options__item\\">23</li><li class=\\"ix-time-options__item\\">24</li><li class=\\"ix-time-options__item\\">25</li><li class=\\"ix-time-options__item\\">26</li><li class=\\"ix-time-options__item\\">27</li><li class=\\"ix-time-options__item\\">28</li><li class=\\"ix-time-options__item\\">29</li><li class=\\"ix-time-options__item\\">30</li><li class=\\"ix-time-options__item\\">31</li><li class=\\"ix-time-options__item\\">32</li><li class=\\"ix-time-options__item\\">33</li><li class=\\"ix-time-options__item\\">34</li><li class=\\"ix-time-options__item\\">35</li><li class=\\"ix-time-options__item\\">36</li><li class=\\"ix-time-options__item\\">37</li><li class=\\"ix-time-options__item\\">38</li><li class=\\"ix-time-options__item\\">39</li><li class=\\"ix-time-options__item\\">40</li><li class=\\"ix-time-options__item\\">41</li><li class=\\"ix-time-options__item\\">42</li><li class=\\"ix-time-options__item\\">43</li><li class=\\"ix-time-options__item\\">44</li><li class=\\"ix-time-options__item\\">45</li><li class=\\"ix-time-options__item\\">46</li><li class=\\"ix-time-options__item\\">47</li><li class=\\"ix-time-options__item\\">48</li><li class=\\"ix-time-options__item\\">49</li><li class=\\"ix-time-options__item\\">50</li><li class=\\"ix-time-options__item\\">51</li><li class=\\"ix-time-options__item\\">52</li><li class=\\"ix-time-options__item\\">53</li><li class=\\"ix-time-options__item\\">54</li><li class=\\"ix-time-options__item\\">55</li><li class=\\"ix-time-options__item\\">56</li><li class=\\"ix-time-options__item\\">57</li><li class=\\"ix-time-options__item\\">58</li><li class=\\"ix-time-options__item\\">59</li></ul><ul class=\\"ix-time-options\\"><li class=\\"ix-time-options__item\\">00</li><li class=\\"ix-time-options__item ix-time-options__item--selected\\">01</li><li class=\\"ix-time-options__item\\">02</li><li class=\\"ix-time-options__item\\">03</li><li class=\\"ix-time-options__item\\">04</li><li class=\\"ix-time-options__item\\">05</li><li class=\\"ix-time-options__item\\">06</li><li class=\\"ix-time-options__item\\">07</li><li class=\\"ix-time-options__item\\">08</li><li class=\\"ix-time-options__item\\">09</li><li class=\\"ix-time-options__item\\">10</li><li class=\\"ix-time-options__item\\">11</li><li class=\\"ix-time-options__item\\">12</li><li class=\\"ix-time-options__item\\">13</li><li class=\\"ix-time-options__item\\">14</li><li class=\\"ix-time-options__item\\">15</li><li class=\\"ix-time-options__item\\">16</li><li class=\\"ix-time-options__item\\">17</li><li class=\\"ix-time-options__item\\">18</li><li class=\\"ix-time-options__item\\">19</li><li class=\\"ix-time-options__item\\">20</li><li class=\\"ix-time-options__item\\">21</li><li class=\\"ix-time-options__item\\">22</li><li class=\\"ix-time-options__item\\">23</li><li class=\\"ix-time-options__item\\">24</li><li class=\\"ix-time-options__item\\">25</li><li class=\\"ix-time-options__item\\">26</li><li class=\\"ix-time-options__item\\">27</li><li class=\\"ix-time-options__item\\">28</li><li class=\\"ix-time-options__item\\">29</li><li class=\\"ix-time-options__item\\">30</li><li class=\\"ix-time-options__item\\">31</li><li class=\\"ix-time-options__item\\">32</li><li class=\\"ix-time-options__item\\">33</li><li class=\\"ix-time-options__item\\">34</li><li class=\\"ix-time-options__item\\">35</li><li class=\\"ix-time-options__item\\">36</li><li class=\\"ix-time-options__item\\">37</li><li class=\\"ix-time-options__item\\">38</li><li class=\\"ix-time-options__item\\">39</li><li class=\\"ix-time-options__item\\">40</li><li class=\\"ix-time-options__item\\">41</li><li class=\\"ix-time-options__item\\">42</li><li class=\\"ix-time-options__item\\">43</li><li class=\\"ix-time-options__item\\">44</li><li class=\\"ix-time-options__item\\">45</li><li class=\\"ix-time-options__item\\">46</li><li class=\\"ix-time-options__item\\">47</li><li class=\\"ix-time-options__item\\">48</li><li class=\\"ix-time-options__item\\">49</li><li class=\\"ix-time-options__item\\">50</li><li class=\\"ix-time-options__item\\">51</li><li class=\\"ix-time-options__item\\">52</li><li class=\\"ix-time-options__item\\">53</li><li class=\\"ix-time-options__item\\">54</li><li class=\\"ix-time-options__item\\">55</li><li class=\\"ix-time-options__item\\">56</li><li class=\\"ix-time-options__item\\">57</li><li class=\\"ix-time-options__item\\">58</li><li class=\\"ix-time-options__item\\">59</li></ul><ul class=\\"ix-time-options\\"><li class=\\"ix-time-options__item ix-time-options__item--selected\\">am</li><li class=\\"ix-time-options__item\\">pm</li></ul>"`;

exports[`TimePicker.vue render work 1`] = `"<div class=\\"ix-time-picker\\"><span class=\\"ix-input ix-input-medium\\"><!--v-if--><span class=\\"ix-input-wrapper\\"><!--v-if--><input placeholder=\\"请选择时间\\" class=\\"ix-input-inner\\"><!--v-if--><span class=\\"ix-input-suffix\\"><span><i class=\\"ix-icon ix-icon-clock-circle\\" role=\\"img\\" arialabel=\\"clock-circle\\"></i></span></span></span><!--v-if--></span><!--teleport start--><!--teleport end--></div>"`;
